packages/react-component-library/src/components/DatePicker/useDatePickerContext.tsx

```diff
@@ -116,10 +154,44 @@ export function useDatePickerReducer(
           endDate: endDate ?? null,
           currentMonth,
           datePickerFormat,
+          isOpen: initialIsOpen,
```

Comment:
IIRC the resetting thing was mainly about when it's a controlled component, and the start date or end date (or date format) change externally (and so the new values need to be brought into the state).

So strictly speaking, I think it should preserve the current `isOpen` value.